src/sql/workbench/contrib/executionPlan/browser/executionPlanPropertiesView.ts

```diff
@@ -109,10 +109,44 @@ export class ExecutionPlanPropertiesView extends ExecutionPlanPropertiesViewBase
 			}
 		];
 
-		this.populateTable(columns, this.convertPropertiesToTableRows(this._model.graphElement?.properties));
+		const tableRows = this.convertPropertiesToTableRows(this._model.graphElement?.properties);
+		this.populateTable(columns, tableRows);
 	}
 
-	private convertPropertiesToTableRows(properties: azdata.executionPlan.ExecutionPlanGraphElementProperty[]): Slick.SlickData[] {
+	public expandOrCollapsePropertiesTable(): void {
```

Comment:
Why is this logic needed? Why can't we just set the property in the base class?

Reply:
I separated the logic, so that expanding and collapsing is a one and done action instead of every time we need to refresh the properties table.